board/xilinx/zynqmp/zynqmp.c

```diff
@@ -344,22 +344,68 @@ int board_early_init_f(void)
 	return 0;
 }
 
-static int multi_boot(void)
+static u32 multi_boot_get(void)
 {
 	u32 multiboot;
 
-	multiboot = readl(&csu_base->multi_boot);
+	zynqmp_mmio_read(&csu_base->multi_boot, &multiboot);
 
-	printf("Multiboot:\t%d\n", multiboot);
+	return multiboot;
+}
+
+static u32 multi_boot_set(u32 multiboot)
+{
+	zynqmp_mmio_write(&csu_base->multi_boot, 0xFFFFFFFF, multiboot);
 
 	return 0;
 }
 
+#ifndef CONFIG_SPL_BUILD
+static int do_multi_boot(struct cmd_tbl *cmdtp, int flag,
+			 int argc, char * const argv[])
+{
+	int ret;
+	u32 multiboot;
+
+	/* If we just want to retrieve current value */
+	if (argc == 1) {
+		multiboot = multi_boot_get();
+
+		printf("Multiboot:\t%d\n", multiboot);
+
+		if (multiboot > 0)
+			ret = env_set("fiovb.is_secondary_boot", 1);
+
+		if (ret)
+			return CMD_RET_FAILURE;
+
+		return CMD_RET_SUCCESS;
+	}
+
+	multiboot = simple_strtoul(argv[1], NULL, 0);
```

Comment:
why not use strict_strtoul and handle the error? this seems a delicate path so the more checks the better

Reply:
Fixed